hooks/cspell/cspell.sh

```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+set -e
+
+git diff --name-only | npx cspell --no-summary --no-progress --no-must-find-files --file-list stdin
```

Comment:
1. Think this is a good stopping point for SPIKE. I think there is definitely a way to setup a shared dictionary which we can determine later.

2. **_Possible limitation_**  with shared dictionaries: In the current implementation we require a dictionary to be present in the repo with the ` pre-commit-config.yaml`. The reason is because this command is run in target repo work directory (it cannot reference files in `pre-commit-hooks`) If we can expose the dictionary some how then it's fine. Otherwise we may want to explore the `cspell` pre-commit hook.

Reply:
Not sure if this makes sense but would it be possible for us to maintain our own cspell config like we do for [commitlint](https://github.com/open-turo/commitlint-config-conventional)  for example, and then that repo would hold our dictionary, and we can just extend it in the local cspell.config.yaml file similar to https://github.com/turo/web-front-end-app/blob/main/.commitlintrc.yaml?

Reply:
Looking at cspell-dicts https://github.com/streetsidesoftware/cspell-dicts/blob/main/dictionaries/ar/README.md, my thought was to try and mimic that - extend/import a shared dictionary that would live in its own repo, and then override this dict as we go in each repo if there are one-off typos that we want to disable for specific repos and not others. If that works I dont think folks would have to touch the shared dictionary frequently (unless they want to introduce a new word that they think shouldnt be a typo across all our codebase).

Reply:
I think the PRs I shared (https://github.com/turo/pre-commit-hooks/pull/44 + https://github.com/turo/web-schumacher-app/pull/9234) approximate what you are describing. We maintain a `turo-dictionary` in this repo that other repos can extend in their `cspell.json` project root file. When you have time, a quick sniff-test of the example PRs would help provide some Platform buy-in, at which point we can modify the RFC accordingly and circulate the content in `#front-end-quality` to get some eyes on it and get the ball rolling. 

(I didn't see `cspell-dicts` though, which is useful: I was only pulling in the dictionaries listed here: https://cspell.org/docs/dictionaries/. Perhaps we can catch more terms with some of the more exhaustive dictionaries. I'll look into it, but the additional benefit at this point is more marginal for each potential dictionary we could add. The integration mostly works.)

Reply:
Just tinkered around with this some more, how do you feel about https://github.com/turo/web-ui/pull/397

Reply:
I think pointing directly to a `.txt` file held in a public repo, even with the extra 3-4 lines of setup in `cspell.json` in usage, is preferable to a dependency listed in `package.json`. When I see a dependency in `package.json` I typically expect there to be some source we're executing. In this case it's just sourcing plain text. The only compelling exception to this opinion would be if we didn't want our typo list to be available publicly.

I think the main point of disagreement is around the following (pulled from the PR description in `web-ui`):

```
But the point being that the shared dictionary doesn't need to live in our pre-commit-hooks repo (in fact I dont think we'd have to go through that repo at all).
```

I think holding the dictionary in `pre-commit-hooks` makes the most sense _precisely_ because it is a dictionary we are pulling into `pre-commit` tooling. Maintaining a separate repo specifically for a dictionary seems more cumbersome than it needs to be, and an example of repo proliferation in the same way we see Slack channel proliferation. Everyone at Turo--front-end and otherwise--is using (or should be using) `pre-commit` tooling. The co-location is helpful in understanding its broader appeal in a way that a stand-alone repo is not so obvious, and is also "closer to the source" (the README in the `pre-commit` repo, in the PR I shared, shows instructions for how to define a `cspell.json` config file, with the only remaining step being adding the `cspell-cli` `pre-commit` hook to the list of hooks the repo is pulling in. That can be added to the README if it would be helpful.)

***

Ultimately, I asked for feedback and got it 😅. I think a next step would be updating the RFC to reflect these two patterns and seeking feedback from the wider team on their preferences.

Reply:
Thanks for all the feedback, just to reiterate my slack DM I'm trying to allocate time to re-immerse myself in this RFC once I have sprint work out of the way 🙏 🙏 Been roughly scanning through these comments though.